`TestWithRsync/with_headless_tsh` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/6406607313/job/17391561212?pr=32920

#### Relevant snippet

```
=== Failed
=== FAIL: tool/tsh/common TestWithRsync/with_headless_tsh (8.67s)
2023-10-04T13:27:18Z WARN [TSH]       "Failed to lock process memory for headless login. Memory locking is used to prevent secrets in memory from being swapped to disk. Please ensure that memory locking is available on your system and your user has locking privileges. This means using a Linux operating system and increasing your user's memory locking limit to unlimited if needed. Alternatively, set --mlock=off or TELEPORT_MLOCK_MODE=off to disable it. This is not recommended in production environments on shared systems where a memory swap attack is possible.\nhttps://goteleport.com/docs/access-controls/guides/headless/#troubleshooting" error:[<nil>] common/tsh.go:4861
2023-10-04T13:27:18Z INFO [CLIENT]    no host login given. defaulting to root client/api.go:1060
2023-10-04T13:27:18Z DEBU [CLIENT]    Skipping connection to the local ssh-agent. client/keyagent.go:137
2023-10-04T13:27:18Z DEBU [TSH]       Pinging the proxy to fetch listening addresses for non-web ports. common/tsh.go:3402
2023-10-04T13:27:18Z DEBU [CLIENT]    attempting to use loopback pool for local proxy addr: 127.0.0.1:34193 client/api.go:4524
2023-10-04T13:27:18Z DEBU [CLIENT]    reading self-signed certs from: /var/lib/teleport/webproxy_cert.pem client/api.go:4532
2023-10-04T13:27:18Z DEBU [CLIENT]    could not open any path in: /var/lib/teleport/webproxy_cert.pem client/api.go:4536
2023-10-04T13:27:18Z DEBU             Attempting GET 127.0.0.1:34193/webapi/ping/headless webclient/webclient.go:129
make: *** [Makefile:785: test-go-tsh] Error 1
make: *** Waiting for unfinished jobs....
2023-10-04T13:27:18Z DEBU             ALPN connection upgrade required for "127.0.0.1:34193": false. client/alpn_conn_upgrade.go:95
2023-10-04T13:27:18Z DEBU [CLIENT]    Activating relogin on no SSH auth methods loaded, are you logged in?. client/api.go:585
2023-10-04T13:27:18Z DEBU [CLIENT]    Attempting to login with a new RSA private key. client/api.go:3859
2023-10-04T13:27:19Z INFO [KEYAGENT]  Loading SSH key for user "alice" and cluster "root". client/keyagent.go:196
2023-10-04T13:27:19Z DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-10-04 14:27:19 +0000 UTC". client/client_store.go:106
2023-10-04T13:27:19Z DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-10-04 14:27:19 +0000 UTC". client/client_store.go:106
2023-10-04T13:27:19Z DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-10-04 14:27:19 +0000 UTC". client/client_store.go:106
2023-10-04T13:27:19Z DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-10-04 14:27:19 +0000 UTC". client/client_store.go:106
2023-10-04T13:27:19Z DEBU [CLIENT]    Device Trust: Skipping device authentication, platform not supported client/api.go:3426
2023-10-04T13:27:19Z DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-10-04 14:27:19 +0000 UTC". client/client_store.go:106
2023-10-04T13:27:19Z DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-10-04 14:27:19 +0000 UTC". client/client_store.go:106
2023-10-04T13:27:19Z DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-10-04 14:27:19 +0000 UTC". client/client_store.go:106
2023-10-04T13:27:19Z DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-10-04 14:27:19 +0000 UTC". client/client_store.go:106
2023-10-04T13:27:19Z DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-10-04 14:27:19 +0000 UTC". client/client_store.go:106
2023-10-04T13:27:19Z DEBU [KEYAGENT]  "Checking key: ssh-rsa-cert-v01@openssh.com AAAAHHNzaC1yc2EtY2VydC12MDFAb3BlbnNzaC5jb20AAAAg4E7kbUYVtBJUFwmM4c7OyMIwm081PmpI4+u5AhabAlAAAAADAQABAAABAQCr++aTlMmBAbZA0BpugpDl+7DnEMZAENODvutyiNhf5lvJ21OZ0nYJA6poKUkpZHb+mstZUEreAhOweTBSHce7zc9iUVVVay+3cGSUddQNHOMeG5Ue4DRFoeCwfDmw3mx8ogWZu1/iyldmI3X1yMsNe/Moi0QtXQGtnxFafEwCDdVW3VPkwOpxJ3rzVWAoohbE8O1NLGelBrJ39PSGgXXtm3UEEnh97AiAE8kJqI6+oHGbbAlr3HQ33ZMQbUsxTxfZzjb/1eBlnHogLcspKW09SCc0l1PAjkngC4Z9udgFL9q1Q21W3D7+ZFJ1FZHm+yffEAHuHMYgnL8LzL/mbK+JAAAAAAAAAAAAAAACAAAAAAAAAJMAAAApMzk5OGM4MWQtZmNlYi00YWRiLTkyNTUtYWRkZmJlN2JmMjliLnJvb3QAAAAkMzk5OGM4MWQtZmNlYi00YWRiLTkyNTUtYWRkZmJlN2JmMjliAAAADXJvb3Rub2RlLnJvb3QAAAAIcm9vdG5vZGUAAAAJbG9jYWxob3N0AAAACTEyNy4wLjAuMQAAAAM6OjEAAAAAZR1n8///////////AAAAAAAAAEMAAAAUeC10ZWxlcG9ydC1hdXRob3JpdHkAAAAIAAAABHJvb3QAAAAPeC10ZWxlcG9ydC1yb2xlAAAACAAAAAROb2RlAAAAAAAAARcAAAAHc3NoLXJzYQAAAAMBAAEAAAEBANF4M6ywPXRDsnrwk1YaDkx9ti4iMupVQ2hNHV9+GQhuf0zDIIIIiZ+QSMHOg7Na3saHh8SHOqAN7G2qA0LEKYYinwkcj55dIZEI9acyWTK/k4KAPhbx7lbDvSBr/LFNUwQWyzx5qq8hQGnXz3hUUd9HwBRvUP8TeoykqydZ8QplDg5tRHS2bYhwv/WrmzCBWEPQdSKIoKCAReYckVxH64Mat8/TEAP3HWd/8b/1QlRVBAc0ISs9qVJ3poTlohwQMcbsOFbSy4YXIxreMOmynZVaExjV6PxiE4Qsoq4QxuUVO/It/27uMXTYtsKWdL5X3V1NWm18Kj0Z+lL5y92Bjf0AAAEUAAAADHJzYS1zaGEyLTUxMgAAAQA+vZV1mqO53lCmBcEm9FUR8DF7rlPWMXMQzeogLZLiIRv70m2QMRo0IpQohJi67orXUpxxNCCDl1tuG0VnxPBCJVnZWFN+vkcd651Y1+ODnc+EvtolYQ6L0QLAnfoUH6LJ0VU6yhH+96vTRhW0wdlta2zyx1ILnOVpgCOaJ0asrGpRnroIROGHebCG2azrB2IZYegkWQ4JoL2YyYsi6EvNYBk3dA8G9hFa5wHvK3KtVZZKqUvY5Q7EaHCT7Zlhxc1CPhGOXJmnVUYHLr93CBITu3Cul8zObWU+BjK/VOlNFgbzUjDOzrXxThQFD+NnN4RST2+6TYCsM9gWULUFIzRX\n." client/keyagent.go:368
2023-10-04T13:27:19Z DEBU [KEYAGENT]  Validated host 127.0.0.1:0@default@root. client/keyagent.go:374
2023-10-04T13:27:20Z DEBU [CLIENT]    No Key Agent selected. client/session.go:278
warning: GOCOVERDIR not set, no coverage data emitted
    proxy_test.go:480: 
        	Error Trace:	/__w/teleport/teleport/tool/tsh/common/proxy_test.go:480
        	Error:      	Received unexpected error:
        	            	signal: killed
        	Test:       	TestWithRsync/with_headless_tsh

```

--- a/tool/tsh/common/proxy_test.go
+++ b/tool/tsh/common/proxy_test.go
@@ -477,14 +477,16 @@ func TestWithRsync(t *testing.T) {
 			require.NoError(t, err)
 			dstPath := filepath.Join(testDir, "dst")
 
-			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
 			t.Cleanup(cancel)
+
 			cmd := tt.createCmd(ctx, testDir, srcPath, dstPath)
-			out, err := cmd.CombinedOutput()
+			err := cmd.Run()
 			var msg string
 			var exitErr *exec.ExitError
 			if errors.As(err, &exitErr) {
-				msg = fmt.Sprintf("exit code: %d\nout: %s", exitErr.ExitCode(), out)
+				msg = fmt.Sprintf("exit code: %d", exitErr.ExitCode())
+				msg += fmt.Sprintf("stderr: %s", exitErr.Stderr)
 			}
 			require.NoError(t, err, msg)
 

